bin/oay/src/services/webdav/service.rs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::convert::Infallible;
+use std::sync::Arc;
+
+use axum::body::Body;
+use axum::http::Request;
+use axum::routing::any_service;
+use axum::Router;
+use dav_server::DavHandler;
+
+use crate::Config;
+
+use super::webdavfs::WebdavFs;
+
+pub struct WebdavService {
+    cfg: Arc<Config>,
+    webdavfs: Box<WebdavFs>,
+}
+
+impl WebdavService {
```

Comment:
It appears that this `main` function is not needed. We should merge it into the `oay.rs` file for startup.